ambry-store/src/main/java/com.github.ambry.store/CompactionManager.java

```diff
@@ -314,11 +315,18 @@ boolean scheduleNextForCompaction(BlobStore store) {
 
     /**
      * Disable the compaction on given blobstore
-     * @param store the {@link BlobStore} on which the compaction is disabled.
+     * @param store the {@link BlobStore} on which the compaction is enabled or disabled.
+     * @param enabled whether to enable ({@code true}) or disable.
      * @return {@code true} if the disabling was successful. {@code false} if not.
      */
-    boolean disableCompactionForBlobStore(BlobStore store) {
-      storesDisabledCompaction.add(store);
+    boolean controlCompactionForBlobStore(BlobStore store, boolean enabled) {
+      if (enabled) {
```

Comment:
`if (enabled && storesDisabledCompaction.contains(store)`
`else if (!enabled)`

Why do you need the `contains` check?

Reply:
Unnecessary `contains` check is removed.